sklearnex/linear_model/linear.py

```diff
@@ -17,365 +17,341 @@
+        assert sample_weight is None
+
+        check_params = {
+            "X": X,
+            "y": y,
+            "dtype": [np.float64, np.float32],
+            "accept_sparse": ["csr", "csc", "coo"],
+            "y_numeric": True,
+            "multi_output": True,
+            "force_all_finite": False,
+        }
+        if sklearn_check_version("1.2"):
+            X, y = self._validate_data(**check_params)
+        else:
+            X, y = check_X_y(**check_params)
 
-            normalize_is_set = (
-                hasattr(self, "normalize")
-                and self.normalize
-                and self.normalize != "deprecated"
-            )
-            positive_is_set = hasattr(self, "positive") and self.positive
-
-            n_samples = _num_samples(X)
-            n_features = _num_features(X, fallback_1d=True)
-
-            # Check if equations are well defined
-            is_good_for_onedal = n_samples > (n_features + int(self.fit_intercept))
-
-            dal_ready = patching_status.and_conditions(
-                [
-                    (sample_weight is None, "Sample weight is not supported."),
-                    (
-                        not issparse(X) and not issparse(y),
-                        "Sparse input is not supported.",
-                    ),
-                    (not normalize_is_set, "Normalization is not supported."),
-                    (
-                        not positive_is_set,
-                        "Forced positive coefficients are not supported.",
-                    ),
-                    (
-                        is_good_for_onedal,
-                        "The shape of X (fitting) does not satisfy oneDAL requirements:."
-                        "Number of features + 1 >= number of samples.",
-                    ),
-                ]
+        if sklearn_check_version("1.0") and not sklearn_check_version("1.2"):
+            self._normalize = _deprecate_normalize(
+                self.normalize,
+                default=False,
+                estimator_name=self.__class__.__name__,
             )
-            if not dal_ready:
-                return patching_status
 
-            if not patching_status.and_condition(
-                self._test_type_and_finiteness(X), "Input X is not supported."
-            ):
-                return patching_status
+        self._initialize_onedal_estimator()
+        try:
+            self._onedal_estimator.fit(X, y, queue=queue)
+            self._save_attributes()
 
-            patching_status.and_condition(
-                self._test_type_and_finiteness(y), "Input y is not supported."
+        except RuntimeError:
+            logging.getLogger("sklearnex").info(
+                f"{self.__class__.__name__}.fit "
+                + get_patch_message("sklearn_after_onedal")
             )
 
-            return patching_status
+            del self._onedal_estimator
+            super().fit(X, y)

```

Comment:
You are not changing flow here. We can leave it as is. But discussion on failed onedal4py run and fallback to stock sklearn is needed.
In my opinion we shouldn't fallback to stock sklearn after dispatching to onedal4py.